### DISCUSSION TITLE
Is it possible to have different template for the same route with the App Directory?

### DISCUSSION THREAD

**Discussion by jpedroribeiro** [vercel/next.js#45949] [Help]
I have the route `/app/blog/[post]/page.tsx` which serves content for blog posts following the pattern: 
- www.example.com/blog/article-a/
- www.example.com/blog/article-b/

However, there are a couple cases where I want to serve a different template whilst keeping this same URL pattern, something like: - www.example.com/blog/different-article-c/
- www.example.com/blog/different-article-d/

I tried a couple of ideas: 
**1)** Hardcoding a new folder for each slug and copy/pasting the template file inside it, like:
- `/blog/different-article-c/page.tsx`
- `/blog/different-article-d/page.tsx`
This is obviously not maintainable.

**2)** Also tried having both templates inside the route `/app/blog/[post]/page.tsx` and testing the slug agains my hardcoded list, but then the output would be bloated with both templates delivered to the browser.

**3)** I could use Middleware to redirect to a different folder. But this feels like pushing the problem somewhere else rather than solving it.

Is there any other way?

**Comment by kfirfitousi** [comment]
I would suggest something like this:

```
app/
├─ blog/
│  ├─ [post]/
│  │  ├─ page.tsx
│  ├─ special/
│  │  ├─ [post]/
│  │  │  ├─ page.tsx
```

And in `page.tsx` use some logic inside `generateStaticParams` to choose which posts to include in the `/special` route.


```tsx
// /app/blog/special/[post]/page.tsx

export function generateStaticParams() {
  const posts = allPosts.filter((post) => post.isSpecial);

  return posts.map(({ slug }) => ({
     post: slug
  });
}
```

```tsx
// /app/blog/[post]/page.tsx

export function generateStaticParams() {
  const posts = allPosts.filter((post) => !post.isSpecial);

  return posts.map(({ slug }) => ({
     post: slug
  });
}
```

**Reply by jpedroribeiro** [reply]
Isn't this still effectively creating an URL like www.example.com/blog/topic/[post]/ ? 
I wanted to avoid having that folder on the final URL and keep them all under `/blog/[post]/`.
